tic-tac-toe-app/Board.jsx

```diff
@@ -0,0 +1,87 @@
+// main board component
+function Board() {
+  // creating state for the board squares
+  const [boardSquares, setBoardSquares] = React.useState(
+    new Array(9).fill(null)
+  );
+
+  // state for the next turn
+  const [isXPlayer, setXPlayer] = React.useState(true);
+
+  // fucntion for when a sqaure is clicked
+  const whenClicked = index => {
+    // spread the items in boardSquare to new array to make a copy
+    const squares = [...boardSquares];
+    // check if square has a value
+    if (squares[index]) return;
+    // add next turn
+    squares[index] = isXPlayer ? 'X' : 'O';
+
+    // set the new state of the board squares
+    setBoardSquares(squares);
+
+    // change the state for the next turn
+    setXPlayer(!isXPlayer);
+  };
+
+  // rendering the squares using square component
+  const renderSquare = index => {
+    return (
+      <Square value={boardSquares[index]} onClick={() => whenClicked(index)} />
+    );
+  };
+
+  // create the gameStatus to determine next player and winner
+  let gameStatus;
+  const winner = findWinner(boardSquares);
+  gameStatus = winner
+    ? `Winner is ${winner}!`
+    : `It is${isXPlayer ? 'X' : 'O'}'s Turn...'`;
+
+  return (
+    <div className="board">
+      <h2 className={gameStatus}>{gameStatus}</h2>
+      <div className="boardRow">
+        {renderSquare(0)}
+        {renderSquare(1)}
+        {renderSquare(2)}
+      </div>
+      <div className="boardRow">
+        {renderSquare(3)}
+        {renderSquare(4)}
+        {renderSquare(5)}
+      </div>
+      <div className="boardRow">
+        {renderSquare(6)}
+        {renderSquare(7)}
+        {renderSquare(8)}
```

Comment:
Creating a `renderSquare()` helper function is a helpful way of reducing duplicate code. 👍 . To kick this reduction of code up a notch, consider refactoring by doing something such as mapping through the `boardSquares` in order to render the instances of `<Square/>`. This would remove the need to render each square separately.